xrpl/models/base_model.py

```diff
@@ -266,20 +266,78 @@ def is_valid(self: BaseModel) -> bool:
         """
         return len(self._get_errors()) == 0
 
-    def _get_errors(self: BaseModel) -> Dict[str, str]:
+    def _check_type(
+        self: Self, attr: str, value: Any, expected_type: Type[Any]
+    ) -> Dict[str, str]:
+        # returns error message if type is bad, None if type is good
+        expected_type_origin = get_origin(expected_type)
+        if expected_type_origin is Union:
+            if any(
+                len(self._check_type(attr, value, expected_type_option)) == 0
+                for expected_type_option in get_args(expected_type)
+            ):
+                return {}
+            return {attr: f"{attr} is {type(value)}, expected {expected_type}"}
+
+        if expected_type is Any:
+            return {}
+
+        if expected_type_origin is list:
+            # expected a List, received a List
+            # TODO: check for
+            if not isinstance(value, list):
+                return {"attr": f"{attr} is {type(value)}, expected {expected_type}"}
+            result = {}
+            for i in range(len(value)):
+                result.update(
+                    self._check_type(
+                        f"{attr}[{i}]", value[i], get_args(expected_type)[0]
```

Comment:
This is actually incorrect - `expected_type` doesn't have the same setup as the list, `get_args(expected_type)[0]` is the subtype of the list.

Reply:
the below flagged example involving `paths=[[PathStep(**path_data) for path_data in path] for path in paths],` in `snippets/paths.py` adheres to your hypothesis because all of the elements have the same type.

But can't we have a `List` of non-identical types in a model? Ex: `List[int, str, float]` ?

Reply:
No, it would be a `List[Union[str, int, float]]`, in which case the child object is still `Union[str, int, float]`. `Union`s are handled in a separate clause.

Reply:
No, I understand Unions. Can't I group attributes of different data types into a `List` inside a model? Ex: `asset3: List[int, str, float]` ?

One could refactor this into three distinct attributes (or perhaps a `Tuple` type). Shouldn't we account for this possibility in the future?

Reply:
If you want to group different data types in a `List`, you'd have to do `List[Union[int, str, float]]`. If you want a `["str", 1, 1.2]` list in specifically that order, then that's a tuple. We don't have any tuple types, so none of the functions (including `from_xrpl`) support it. It won't be hard to add support for tuples if we do need tuple models, but I don't see that happening since rippled's serialization doesn't support that.

Reply:
I see, okay. that makes sense